ipfs/ipfs_utils.go

```diff
@@ -0,0 +1,107 @@
+package ipfs
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+)
+
+func (c *client) getDataDir(network string) string {
+	p, _ := filepath.Abs(filepath.Join(c.dataDir, fmt.Sprintf("/data/ipfs/%s", network)))
+	return p
+}
+
+func (c *client) waitForNode(ctx context.Context, dockerID string) error {
+	logs, err := c.d.ContainerLogs(ctx, dockerID, types.ContainerLogsOptions{
+		ShowStdout: true,
+		Follow:     true,
+	})
+	if err != nil {
+		return err
+	}
+	defer logs.Close()
+
+	scanner := bufio.NewScanner(logs)
+	for scanner.Scan() {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("cancelled wait for %s", dockerID)
+		default:
+			if strings.Contains(scanner.Text(), "Daemon is ready") {
+				return nil
+			}
+		}
+	}
+
+	return scanner.Err()
+}
+
+func (c *client) initNodeFS(n *NodeInfo, opts NodeOpts) error {
+	// set up directories
+	os.MkdirAll(c.getDataDir(n.NetworkID), c.fileMode)
+
+	// write swarm.key to mount point, otherwise check if a swarm key exists
+	keyPath := c.getDataDir(n.NetworkID) + "/swarm.key"
+	if opts.SwarmKey != nil {
```

Comment:
Nevermind i was missing something, looks like the key is writen here.

Reply:
ah my bad, the naming isn't very clear - i'll improve the docs and function name